spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/DatabaseType.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.sql;
+
+/**
+ * @author João André Martins
+ */
+public enum DatabaseType {
+	MYSQL("mysql", "com.mysql.jdbc.Driver", "jdbc:mysql://google/%s?cloudSqlInstance=%s&"
+			+ "socketFactory=com.google.cloud.sql.mysql.SocketFactory&useSSL=false"),
+	POSTGRESQL("postgresql", "org.postgresql.Driver", "jdbc:postgresql://google/%s?"
+			+ "socketFactory=com.google.cloud.sql.postgres.SocketFactory&socketFactoryArg=%s"
+			+ "&useSSL=false");
+
+	private String name;
+
+	private String jdbcDriverName;
+
+	private String jdbcUrlTemplate;
```

Comment:
All properties must be `final` and there should be a blank line between enum values.
I think we should configure some Checkstyle rule on the matter. Don't know which one though...